src/main/java/org/wso2/carbon/connector/SplitFile.java

```diff
@@ -84,26 +84,27 @@ public void connect(MessageContext messageContext) {
      * @param fileLocation   Location of the source file.
      * @param chunkSize      size of the chunks to split the file.
      * @param destination    Location of the destination to write the splitted files.
-     * @param options        Init configuration options.
      * @param messageContext Message context.
      * @return Status true/false.
      */
     private boolean splitFile(String fileLocation, String chunkSize, String destination, String splitLength,
-                              String xpathExpression, FileSystemOptions options, MessageContext messageContext) {
+                              String xpathExpression, MessageContext messageContext) {
         FileObject sourceFileObj = null;
         try {
             manager = FileConnectorUtils.getManager();
-            sourceFileObj = manager.resolveFile(fileLocation, options);
+            FileSystemOptions sourceFso = FileConnectorUtils.getFso(messageContext, fileLocation, manager);
+            FileSystemOptions destinationFso = FileConnectorUtils.getFso(messageContext, fileLocation, manager);
```

Comment:
FileConnectorUtils.getFso(messageContext, fileLocation, manager) should be FileConnectorUtils.getFso(messageContext, destination, manager) right?

Reply:
yes. fixed.